torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -25,47 +28,354 @@ glow::Type ptTypeToGlowType(const at::CompleteTensorType &ptType) {
+  (void)transposedHandle;
+
+  glow::Handle<int32_t> groupsHandle =
+      getGlowConstantHandle<int32_t>(inputs[8]);
+  assert(groupsHandle.size() == 1);
+  uint32_t groups = groupsHandle.raw(0);
+
+  std::vector<uint32_t> kernels = {static_cast<uint32_t>(weightsShape.h),
+                                   static_cast<uint32_t>(weightsShape.w)};
+
+  auto outSz = glow::calculateConvPoolOutputDims(
+      inputShape.h, inputShape.w, kernels, strides, pads, dilation);
+  std::array<size_t, 4> outDims = {
+      {input.dims()[0], outSz.first, outSz.second, weightsShape.n}};
+  glow::TypeRef outTy =
+      f_->getParent()->uniqueType(glow::ElemKind::FloatTy, outDims);
+
+  glow::ConvolutionNode *conv =
+      f_->createConv("conv", input, weights, bias, outTy, kernels, strides,
+                     pads, groups, dilation);
+  glow::TransposeNode *output = f_->createTranspose(
+      "conv_input_transposed", conv->getResult(), NHWC2NCHW);
+
+  addGlowNodeValue(outputs[0], output->getResult());
+}
+
+void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+  assert(inputs.size() == 0);
+  assert(outputs.size() == 1);
+
+  auto optionalIValue = torch::jit::toIValue(outputs[0]);
+  assert(optionalIValue.has_value() && "Constants should have IValue outputs");
+  const torch::jit::IValue iVal = *optionalIValue;
+
+  glow::Module *mod = f_->getParent();
+
+  glow::Tensor t;
+
+  if (iVal.isInt()) {
+    int64_t val = iVal.toInt();
+    t.reset(glow::ElemKind::Int32ITy, {1});
+    t.getHandle<int32_t>().raw(0) = to32Bit(val);
+  } else if (iVal.isIntList()) {
+    const auto vals = iVal.toIntListRef();
+    t.reset(glow::ElemKind::Int32ITy, {vals.size()});
+    auto handle = t.getHandle<int32_t>();
+    for (size_t i = 0; i < vals.size(); ++i) {
+      handle.raw(i) = to32Bit(vals[i]);
+    }
+  } else if (iVal.isDouble()) {
+    double val = iVal.toDouble();
+    t.reset(glow::ElemKind::FloatTy, {1});
+    t.getHandle<float>().raw(0) = to32Bit(val);
+  } else if (iVal.isDoubleList()) {
+    const auto vals = iVal.toDoubleListRef();
+    t.reset(glow::ElemKind::FloatTy, {vals.size()});
+    auto handle = t.getHandle<float>();
+    for (size_t i = 0; i < vals.size(); ++i) {
+      handle.raw(i) = to32Bit(vals[i]);
+    }
+  } else if (iVal.isBool()) {
+    bool val = iVal.toBool();
+    t.reset(glow::ElemKind::BoolTy, {1});
+    t.getHandle<bool>().raw(0) = val;
+  } else if (iVal.isBoolList()) {
+    const auto &valsList = iVal.toBoolList();
+    const auto &vals = c10::impl::toVector(valsList);
+    t.reset(glow::ElemKind::BoolTy, {vals.size()});
+    auto handle = t.getHandle<bool>();
+    for (size_t i = 0; i < vals.size(); ++i) {
+      handle.raw(i) = vals[i];
+    }
+  } else if (iVal.isTensor()) {
+    assert(false && "not yet implemented tensor constants");
+  } else if (iVal.isNone()) {
+    // Nothing to do for None
```

Comment:
Do we need to initialize t to be none here?

Reply:
Basically we are handling None as an absence a of Glow::NodeValue so when we see one, we want to do nothing but it's still valid to see one

Reply:
Yeah, I mean t is not intialized here, is it okay?

Reply:
Yes it's on purpose, Glow has no equivalent of python `None`